internal/controllers/addon/controller.go

```diff
@@ -137,7 +140,14 @@ func (r *AddonReconciler) Reconcile(
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	isUninstalling := false
 	defer func() {
+		// Update metrics only if a Recorder is initialized
+		if r.Recorder != nil {
+			r.Recorder.HandleAddonConditionAndInstallCount(string(addon.UID),
```

Comment:
Why not hand in the entire Addon object here?
This way we don't need this `isUninstalling` variable in as the `HandleAddonConditionAndInstallCount` method can also check for `.DeletionTimestamp`.

`HandleAddonConditionAndInstallCount` could also just become `RecordAddonMetrics` or is there a reason for multiple entrypoint on the `*Recorder` type to record Addon Metrics?